lib/ui/painting.dart

```diff
@@ -1196,6 +1196,16 @@ class Paint {
   }
 }
 
+/// The encoding formats supported by the [Image.toByteData].
+// These enum values must be kept in sync with the logic in ToSkEncodedImageFormat.
+enum EncodeFormat {
+  // Be conservative with the formats we expose. It is easy to add new formats in future but
+  // difficult to remove.
+  JPEG,
+  PNG,
+  WEBP,
+}
```

Comment:
These should be lowercase.

Here are some doc strings for each one:
```
/// JPEG format.
///
/// This format, strictly speaking called JFIF, is a lossy compression graphics format that
/// can handle images up to 65,535 pixels in either dimension. The `quality` metric controls
/// the compression ratio. Values in the range of about 50 to 90 are somewhat reasonable;
/// values above 95 increase the file size with little noticeable improvement to the quality,
/// values below 50 drop the quality substantially.
///
/// This format is well suited for photographs. It is very poorly suited for images with hard
/// edges or text. It does not support transparency.
///
/// JPEG images normally use the `.jpeg` file extension and the `image/jpeg` MIME type.
///
/// See also:
///
///  * <https://en.wikipedia.org/wiki/JPEG>, the Wikipedia page on JPEG.
jpeg,

/// PNG format.
///
/// A loss-less compression format for images. This format is well suited for images
/// with hard edges, such as screenshots or sprites, and images with text. Transparency
/// is supported. The `quality` metric is ignored for this format. The PNG format supports
/// images up to 2,147,483,647 pixels in either dimension, though in practice available
/// memory provides a more immediate limitation on maximum image size.
///
/// PNG images normally use the `.png` file extension and the `image/png` MIME type.
///
/// See also:
///
///  * <https://en.wikipedia.org/wiki/Portable_Network_Graphics>, the Wikipedia page on PNG.
///  * <https://tools.ietf.org/rfc/rfc2083.txt>, the PNG standard.
png,

/// WebP format.
///
/// The WebP format supports both lossy and lossless compression; however, the
/// [Image.toByteData] method always uses lossy compression when [webp] is
/// specified. The `quality` metric determines the compression ratio; higher values
/// result in better quality but larger file sizes, and vice versa. WebP images are
/// limited to 16,383 pixels in each direction (width and height).
///
/// WebP images normally use the `.webp` file extension and the `image/webp` MIME type.
///
/// See also:
///
///  * <https://en.wikipedia.org/wiki/WebP>, the Wikipedia page on WebP.
webp,
```

We should figure out